phoenix-core/src/main/java/org/apache/phoenix/mapreduce/index/IndexUpgradeTool.java

```diff
@@ -0,0 +1,459 @@
+            "[Required]Operation to perform (upgrade/rollback)");
+    private static final Option TABLE_OPTION = new Option("tb", "table", true,
+            "[Required]Tables and indexes list ex. table1: index1, index2, index3; table2: index4, index5");
+    private static final Option TABLE_CSV_FILE_OPTION = new Option("f", "file", true,
+            "[Optional]Tables and indexes list in a csv file");
+    private static final Option DRY_RUN_OPTION = new Option("s", "show", false,
+            "[Optional]If passed this will output steps that will be executed");
+    private static final Option HELP_OPTION = new Option("h", "help", false, "Help");
+    private static final Option CONNECTION_STRING_OPTION = new Option("c", "connection-string", true, "Connection string");
+    private static final Option LOG_FILE_OPTION = new Option("lf", "logfile", true,
+            "Log file path where the logs are written");
+    private static final Option HOST_OPTION = new Option("ho", "host", true, "[Required]zookeeper quorum host");
+    private static final Option PORT_OPTION = new Option("po", "port", true, "[Required]zookeeper port");
+
+    private static final String UPGRADE_OP = "upgrade";
+    private static final String ROLLBACK_OP = "rollback";
+
+
+    private static HashMap <String, Boolean> tableMC = new HashMap<>();
+    private static Map<String, String> prop = new  HashMap<>();
+
+    private static String connectionURL;
+    private static boolean dryRun;
+
+    public static void main (String[] args) {
+        CommandLine cmdLine = null;
+        try {
+            try {
+                cmdLine = parseOptions(args);
+            } catch (IllegalStateException e) {
+                printHelpAndExit(e.getMessage(), getOptions());
+            }
+            String operation = cmdLine.getOptionValue(OPERATION_OPTION.getOpt());
+            String tables = cmdLine.getOptionValue(TABLE_OPTION.getOpt());
+            String host = cmdLine.getOptionValue(HOST_OPTION.getOpt());
+            String port = cmdLine.getOptionValue(PORT_OPTION.getOpt());
+            dryRun = Boolean.valueOf(cmdLine.getOptionValue(DRY_RUN_OPTION.getOpt()));
+
+            String logFile = cmdLine.getOptionValue(LOG_FILE_OPTION.getOpt());
+            if (logFile != null) {
+                FileHandler fh = new FileHandler(logFile);
+                fh.setFormatter(new SimpleFormatter());
+                LOGGER.addHandler(fh);
+            }
+            if (dryRun) {
+                LOGGER.info("This is the beginning of the tool with dry run.");
+            }
+            LOGGER.info("Index Upgrade tool initiated: "+ String.join(",", args));
+
+            prop.put(Indexer.INDEX_BUILDER_CONF_KEY, PhoenixIndexBuilder.class.getName());
+            prop.put(NonTxIndexBuilder.CODEC_CLASS_NAME_KEY, PhoenixIndexCodec.class.getName());
+
+            connectionURL = cmdLine.getOptionValue(CONNECTION_STRING_OPTION.getOpt());
```

Comment:
rather than a connection string, it's easier on the end user to take a ZK quorum. We can create a connection string of our own from a ZK quorum using ConnectionUtil.

Reply:
Yes, removed this option.